site/_js/web-components/load-more.js

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {BaseElement} from './base-element';
+import {html} from 'lit-element';
+import {unsafeHTML} from 'lit-html/directives/unsafe-html';
+import {unsafeSVG} from 'lit-html/directives/unsafe-svg';
+import arrowIcon from '../../_includes/icons/arrow-forward.svg';
+import {generateIdSalt} from '../utils/salt';
+
+export class LoadMore extends BaseElement {
+  constructor() {
+    super();
+
+    this.total = null;
+    this.skip = 0;
+    this.take = 10;
+    this.fetchItems = null;
+    this.initialItems = [];
+    this.loadedItems = [];
+    this._loading = false;
+    this._haveError = false;
+    this.i18n = {};
+    this._id = `load-more-${generateIdSalt('load-more-')}`;
+  }
+
+  static get properties() {
+    return {
+      skip: {type: Number, reflect: true},
+      take: {type: Number, reflect: true},
+      total: {type: Number, reflect: true},
+      fetchItems: {type: Function, reflect: false},
```

Comment:
Please add jsdoc for the signature of fetchItems (arguments skip, take and that it expects full-html items back)